python/paddle/fluid/executor.py

```diff
@@ -610,3 +611,201 @@ def _run(self, program, exe, feed, fetch_list, feed_var_name,
 
     def _run_inference(self, exe, feed):
         return exe.run(feed)
+
+    def _dump_debug_info(self, program=None, trainer=None):
+        with open(str(id(program)) + "_train_desc.prototxt", "w") as fout:
+            fout.write(trainer._desc())
+        if program._fleet_opt:
+            with open("fleet_desc.prototxt", "w") as fout:
+                fout.write(str(program._fleet_opt["fleet_desc"]))
+
+    def _prepare_trainer(self,
+                         program=None,
+                         dataset=None,
+                         scope=None,
+                         thread=0,
+                         debug=False,
+                         fetch_list=None,
+                         fetch_info=None,
+                         print_period=100):
+        if scope is None:
+            scope = global_scope()
+        if fetch_list is None:
+            fetch_list = []
+        if fetch_info is None:
+            fetch_info = []
+        assert len(fetch_list) == len(fetch_info)
+        compiled = isinstance(program, compiler.CompiledProgram)
+        if not compiled:
+            trainer = TrainerFactory()._create_trainer(program._fleet_opt)
+            trainer._set_program(program)
+        else:
+            trainer = TrainerFactory()._create_trainer(
+                program.program._fleet_opt)
+            trainer._set_program(program.program)
+        if thread <= 0:
+            if dataset.thread_num <= 0:
+                raise RuntimeError(
+                    "You should set thread num first, either in Dataset"
+                    "or in Executor.train_from_dataset")
+            else:
+                trainer._set_thread(dataset.thread_num)
+        else:
+            trainer._set_thread(thread)
+        trainer._set_debug(debug)
+        trainer._set_fetch_var_and_info(fetch_list, fetch_info, print_period)
+        return scope, trainer
+
+    def infer_from_dataset(self,
+                           program=None,
+                           dataset=None,
+                           scope=None,
+                           thread=0,
+                           debug=False,
+                           fetch_list=None,
+                           fetch_info=None,
+                           print_period=100):
+        """
+        The document of infer_from_dataset is almost the same as
```

Comment:
please preview infer_from_dataset

Reply:
![image](https://user-images.githubusercontent.com/35550832/55240892-dbdeb380-5274-11e9-92fa-f6e8a5f177dd.png)
![image](https://user-images.githubusercontent.com/35550832/55240930-f31da100-5274-11e9-8e2a-141052c11786.png)

Reply:
![image](https://user-images.githubusercontent.com/35982308/55273104-679e2180-5301-11e9-8d8b-d2992c5e16a5.png)

Reply:
fixed.

Reply:
![image](https://user-images.githubusercontent.com/35550832/55277721-2b3ce680-533e-11e9-9d5d-669ec88ee785.png)
![image](https://user-images.githubusercontent.com/35550832/55277724-3132c780-533e-11e9-9762-17b9a90d526e.png)

Reply:
Hi, can you recover the api format to the previous version, that is :
![image](https://user-images.githubusercontent.com/31058429/55284809-a9929a80-53c9-11e9-893f-73a74f2b1c74.png)

As for the :param program: , please treat it in the same way as other parameters so that the problem can be fixed.

An ideal api format of a class method please refer to:
http://www.paddlepaddle.org/documentation/docs/en/1.3/api/data_feeder.html

Thanks.